### PR TITLE
upload example code for snail with gluon

### PR DESCRIPTION
**Pull request by seujung**
## Description ##
- upload example for snail code
   - SNAIL(A Simple Neural Attentive Meta-Learniner) is a kind of meta-learning algorithm.
   - Train network using omniglot dataset

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: example/gluon/snail/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+##### network structore
+![net_structure](assets/net_structure.png)
+
+##### building block structure
+![block_structure](assets/blocks.png)
+
+## Requirements
+- Python 3.6.1
+- Mxnet 1.2.0
+- pytorch 0.4.0 (only use data loading)
```
Comment (marcoabreu) [inline example/gluon/snail/README.md]: Could you elaborate on that requirement? I can't find a code-reference to pytorch in your PR (besides the inspiration)

**Comment by marcoabreu** [comment]
Thanks a lot for your contribution, this looks great!

File: example/gluon/snail/README.md (hunk 1 above)
Comment (seujung) [reply example/gluon/snail/README.md]: Fix requirements
- delete pytorch
- add mxboard

**Comment by vandanavk** [comment]
Please execute pylint on these files using ci/other/pylintrc in incubator-mxnet folder.
Something like this
pylint --rcfile=ci/other/pylintrc --ignore-patterns="._.so$$,._.dll$$,._.dylib$$" example/gluon/snail/*.py to make sure there are no lint errors.

**Comment by seujung** [comment]
complete pylint test and upload code

**Comment by ankkhedia** [comment]
@ThomasDelteil ping for review!

File: example/gluon/snail/omniglot_dataset.py [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Description:
+Inspired by https://github.com/pytorch/vision/pull/46
```
Comment (anirudhacharya) [inline example/gluon/snail/omniglot_dataset.py]: can you give the reference to actual dataset or code in a github repo instead of a pull request link.

**Comment by anirudhacharya** [comment]
why the change in the sn_gan tutorial?

File: example/gluon/snail/omniglot_dataset.py (hunk 2 above)
Comment (seujung) [reply example/gluon/snail/omniglot_dataset.py]: Update the reference site 
https://github.com/apache/incubator-mxnet/pull/13966/commits/28740d2a1661d17e4f1b89c3c2a5052cdad97fd0

**Comment by seujung** [comment]
For the push issue, I close the Pull Request.
I make a new pr #14367
